python/dgl/sampling/neighbor.py

```diff
@@ -186,39 +186,45 @@ def sample_neighbors(g, nodes, fanout, edge_dir='in', prob=None, replace=False,
                             tag_offset_name='_TAG_OFFSET', replace=False,
                             copy_ndata=True, copy_edata=True):
-    """Sample neighboring edges of the given nodes and return the induced subgraph, where each
-       neighbor's probability to be picked is determined by its tag.
+    r"""Sample neighboring edges of the given nodes and return the induced subgraph, where each
+    neighbor's probability to be picked is determined by its tag.
 
     For each node, a number of inbound (or outbound when ``edge_dir == 'out'``) edges
     will be randomly chosen.  The graph returned will then contain all the nodes in the
     original graph, but only the sampled edges.
 
     This version of neighbor sampling can support the scenario where adjacent nodes with different
-    types might have different probability to be picked. Each node is assigned an integer(tag)
+    types have different sampling probability. Each node is assigned an integer (called a *tag*)
     which represents its type. Tag is an analogue of node type under the framework of homogeneous
     graphs. Nodes with the same tag share the same probability.
 
-    For example, assume a node has (a+b) neighbors, and a of them have tag 0 while b of them have
-    tag 1. Assume a node of tag 0 has an unnormalized probability p to be picked while a node of
-    tag 1 has q. This function first chooses a tag according to the unnormalized probability
-    distribution (ap, bq), and then run a uniform sampling within the nodes with the chosen tag.
-
-    In order to sample efficiently, we need to first sort the CSR matrix of the graph
-    according to the tag (See `dgl.transform.sort_in_edges` and `dgl.transform.sort_out_edges`
-    for details), which will arrange the neighbors with the same tag in a consecutive range
-    and store the offset of these ranges in a node feature with tag_offset_name as its name.
-
-    Please make sure that the graph has been sorted by the sorting function corresponding to
-    the edge direction ('in' or 'out'). This function itself will not check whether the graph is
-    sorted. Note that the input `tag_offset_name` should be consistent with that in the sorting
-    function.
-
-    Only homogeneous or bipartite graphs are supported. For bipartite graphs, only candidate
-    frontier nodes have tags(source nodes when edge_dir='in' and destination nodes when
-    edge_dir='out'), and the offset of tags should be stored as a node feature of the seed nodes.
+    For example, assume a node has :math:`N+M` neighbors, and :math:`N` of them
+    have tag 0 while :math:`M` of them have tag 1. Assume a node of tag 0 has
+    an unnormalized probability :math:`p` to be picked while a node of tag 1
+    has :math:`q`. This function first chooses a tag according to the
+    unnormalized probability distribution
+    :math:`\frac{P(tag=0)}{P(tag=1)}=\frac{Np}{Mq}`, and then run a uniform
```

Comment:
non-uniform sampling with the computed probability.

Reply:
The algorithm has two steps. The first is a non-uniform sampling to get the tag, the second is a uniform sampling to get a node from the nodes of the chosen tag.